include/sparrow/layout/primitive_array.hpp

```diff
@@ -14,13 +14,22 @@
 
 #pragma once
 
+#include <cstddef>
+
 #include "sparrow/arrow_array_schema_proxy.hpp"
+#include "sparrow/buffer/buffer_adaptor.hpp"
 #include "sparrow/layout/array_base.hpp"
 #include "sparrow/utils/iterator.hpp"
 #include "sparrow/utils/nullable.hpp"
 
 namespace sparrow
-{   
+{
+    class run_end_encoded_array;
```

Comment:
This is not required, the friend declaration acts as a forward declaration (local to the class scope).

Reply:
removed